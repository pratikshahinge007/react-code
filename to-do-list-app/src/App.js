import React, { useState } from "react";
import "./style.css";
import logo from './logo.svg'
export default function App() {
  const [inputData, setInputData] = useState("");
  const [finalToDoList, setFinalToDoList] = useState([]);

  const onChangeInput = (e) => {
    setInputData(e.target.value);
  };

  const addToDoListData = () => {
    setFinalToDoList([...finalToDoList, inputData]);
  };

  const handleKeyPress = (e) => {
    if (e.key === "Enter") {
      addToDoListData();
    }
  };

  return (
    <div id="container">
      <div id="item">
        <div id="outer_div">
          <label htmlFor="input">Input</label>
          <input
            id="input"
            type="text"
            value={inputData}
            onChange={onChangeInput}
            onKeyPress={handleKeyPress}
            placeholder="Enter a new task"
          />
          <button id="add_button" onClick={addToDoListData}>
            Add
          </button>
        </div>
        <div id="list_data">
          {finalToDoList?.map((data, id) => {
            return <div key={id}>{data}</div>;
          })}
        </div>
      </div>
      <div id="item">
        <div>to-Do List</div>
        <div>with JavaScript</div>
        <img title="React logo" src={logo.svg}/>
      </div>
    </div>
  );
}
