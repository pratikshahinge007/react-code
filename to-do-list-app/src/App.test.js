import { fireEvent, render,screen } from "@testing-library/react";
import App from './App';

test("Testing To do list", ()=>{
  render(<App/>)
  const toDoText = screen.getByText(/To-Do List/i);
  expect(toDoText).toBeInTheDocument()
  const title = screen.getByTitle("React logo")
  expect(title).toBeInTheDocument()
})
test("Testing To do list", ()=>{
  render(<App/>)
  const toDoText = screen.getByText(/To-Do List/i);
  const title = screen.getByTitle("React logo")
  const toDoText2 = screen.getByText("with JavaScript")
  expect(toDoText).toBeInTheDocument()
  expect(title).toBeInTheDocument()
  expect(toDoText2).toBeInTheDocument()
})

test("textbox is present", () => {
  render(<App/>)
  const textboxPresent = screen.getByRole('textbox')
  expect(textboxPresent).toBeInTheDocument()
})


test('matches the snapshot', () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test('test placeholder text', () => {
  render(<App/>)
  const placeholder = screen.getByPlaceholderText("Enter a new task")
  expect(placeholder).toBeInTheDocument()
})

// test("test label", () => {
//   render(<App/>)
//   const label = screen.getByLabelText(/input_data/i)
//   expect(label).toBeInTheDocument()
// })

// test("test form submission", () => {
//   render(<App/>)

//   const input = screen.getByLabelText(/input_data/i)
//   fireEvent.change(input, {target :{value :"John"}}) 
//   expect(input.value).toBe("John")

//   const button = screen.getByRole("button", {name : /Submit/i})
//   fireEvent.click(button)
//   expect("form submitted").toBeInTheDocument()

//   const submissionMessage = screen.getByText("submitted Successfully")
//   expect(submissionMessage).toBeInTheDocument()
// })



