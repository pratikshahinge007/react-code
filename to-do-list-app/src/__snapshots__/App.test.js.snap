// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    id="container"
  >
    <div
      id="item"
    >
      <div
        id="outer_div"
      >
        <label
          for="input"
        >
          Input
        </label>
        <input
          id="input"
          placeholder="Enter a new task"
          type="text"
          value=""
        />
        <button
          id="add_button"
        >
          Add
        </button>
      </div>
      <div
        id="list_data"
      />
    </div>
    <div
      id="item"
    >
      <div>
        to-Do List
      </div>
      <div>
        with JavaScript
      </div>
      <img
        title="React logo"
      />
    </div>
  </div>
</DocumentFragment>
`;
